R/package.R

```diff
@@ -22,24 +22,26 @@ pkg_install <- function(pkg, lib = .libPaths()[[1L]], num_workers = 1L) {
 
   # Do we need to do anything?
   if (nrow(needs_install) == 0) {
+    plan$num_deps <- total_num_deps(plan)
     dir <- plan[plan$direct, ]
     ind <- plan[!plan$direct, ]
     for (i in seq_len(nrow(dir))) {
       msg_success("{fmt_pkg(dir$package[i])} {fmt_ver(dir$version[i])} \\
-                   already installed")
+                   and {dir$num_deps[i]} dependencies already installed")
     }
-    if (nrow(ind)) msg_success("{nrow(ind)} dependencies already installed")
-    return(invisible(plan))
-  }
 
-  # Remove already installed dependencies from the plan
-  installed <- plan$package[plan$type == "installed"]
-  needs_install$dependencies <-
-    lapply(needs_install$dependencies, setdiff, y = installed)
+  } else {
+    # Remove already installed dependencies from the plan
+    installed <- plan$package[plan$type == "installed"]
+    needs_install$dependencies <-
+      lapply(needs_install$dependencies, setdiff, y = installed)
+
+    # Install what is left
+    install_packages(needs_install$file, lib = lib, plan = needs_install,
+                     num_workers = num_workers)
+  }
 
-  # Install what is left
-  install_packages(needs_install$file, lib = lib, plan = needs_install,
-                   num_workers = num_workers)
+  invisible(plan)
```

Comment:
So `install_packages()` returns data about what was installed and how long it took to install, which is why this was returning 

Maybe it makes sense for pkgman to always just return the installation plan in all cases instead, I am not sure.

Reply:
That sounds good to me!

Reply:
So, I think that `pkg_install` could return the plan, plus the data about the installation. Is that good?

Reply:
It should probably even include the already installed packages...

Reply:
OK, I did it somewhat differently, without modifying `pkginstall`, because `pkginstall::install_packages()` has different return values depending on whether it has a plan or not, etc.

Anyway, now `pkg_install` always returns the plan. It fills in the install locations from the `install_packages()` result.

It also prints the result, so the total time of the install phase is printed.

Reply:
yeah, that makes sense.